dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -0,0 +1,186 @@
+import { css } from '@emotion/react';
+import { useEffect, useState } from 'react';
+import { useApi } from '../lib/useApi';
+
+type Props = {
+	pageId: string;
+	webTitle: string;
+	ajaxUrl: string;
+};
+
+// TODO: Break this out into its own component
+const Toast = ({
+	onClick,
+	noOfNewPosts,
+}: {
+	onClick: () => void;
+	noOfNewPosts: number;
+}) => {
+	// TODO: Style and absolute position this component
+	return (
+		<nav
+			css={css`
+				position: fixed;
+				top: 20px;
+				left: 50px;
+				display: flex;
+				width: 100%;
+				align-items: center;
+			`}
+		>
+			<button onClick={onClick}>{`${noOfNewPosts} blah`}</button>
+		</nav>
+	);
+};
+
+const isServer = typeof window === 'undefined';
+
+function hydrateBlocks() {
+	console.log('hydrateBlocks');
+	// TODO Call existing hydration solution targetted to these elements
+	// Maybe we can make doHydration work for all gu-islands things
+	// where the gu-hydrated data attribute isn't set?
+}
+
+/**
+ * insertNewBlocks takes html and inserts it at the top of the liveblog
+ *
+ * @param {string} html The block html to be inserted
+ * @returns void
+ */
+function insertNewBlocks(html: string) {
+	const latestBlock = document.querySelector('#maincontent :first-child');
+	if (!latestBlock) return;
+	latestBlock.insertAdjacentHTML('beforebegin', html);
+}
+
+/**
+ * revealNewBlocks - style any blocks that have been inserted but are hidden such that
+ * they are revealed
+ */
+function revealNewBlocks() {
+	console.log('revealNewBlocks');
+}
+
+/**
+ * This abstracts the code to construct the url we use for polling. Of note is the fact
+ * we have put `latestBlockId` in react state; by doing this we are able to change the url
+ * that SWR uses for the request whilst still benefitting from the polling logic
+ *
+ * @returns string
+ */
+function getKey(
+	pageId: string,
+	ajaxUrl: string,
+	filterKeyEvents: boolean,
+	latestBlockId: string,
+): string {
+	// Construct the url to poll
+	const url = new URL(`${pageId}.json`, ajaxUrl);
+	url.searchParams.set('lastUpdate', latestBlockId);
+	url.searchParams.set('isLivePage', 'true');
+	url.searchParams.set('filterKeyEvents', filterKeyEvents ? 'true' : 'false');
+	return url.href;
+}
+
+const topOfBlog: Element | null | false =
+	!isServer && window.document.querySelector('[data-gu-marker=top-of-blog]');
+
+/**
+ * This allows us to make decisions in javascript based on if the reader
+ * has the top of the blog in view or not
+ *
+ * @returns boolean
+ */
+function topOfBlogVisible(): boolean {
+	return !!topOfBlog && !!topOfBlog.classList.contains('in-viewport');
+}
+
+export const Liveness = ({ pageId, webTitle, ajaxUrl }: Props) => {
+	const [showToast, setShowToast] = useState(false);
+	const [noOfNewPosts, setNoOfNewPosts] = useState(0);
```

Comment:
noOfNewPosts vs numNewBlocks is a bit confusing. Am I correct in thinking that:
noOfNewPosts = cumulative count of new blocks that are not currently visible on the page 
numNewBlocks = latest number of new blocks retrieved by the latest polling request

If so maybe could they be named:
noOfNewPosts -> cumulativeNewBlocksCount
numNewBlocks -> latestNewBlocksCount
Or something like this?

Reply:
How about:

`noOfNewPosts ` > `numHiddenBlocks`
Keep `numNewBlocks` as it is

?